neo/Network/P2P/Payloads/TransactionAttribute.cs

```diff
@@ -1,45 +1,76 @@
-        public byte[] Data;
+        public readonly TransactionAttributeUsage Usage;
 
-        public int Size => sizeof(TransactionAttributeUsage) + Data.GetVarSize();
+        public virtual int Size => sizeof(TransactionAttributeUsage);
 
-        void ISerializable.Deserialize(BinaryReader reader)
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="usage">Usage</param>
+        protected TransactionAttribute(TransactionAttributeUsage usage)
         {
-            Usage = (TransactionAttributeUsage)reader.ReadByte();
-            if (!Enum.IsDefined(typeof(TransactionAttributeUsage), Usage))
-                throw new FormatException();
-            Data = reader.ReadVarBytes(252);
+            Usage = usage;
         }
 
-        void ISerializable.Serialize(BinaryWriter writer)
+        protected abstract void DeserializeWithoutType(BinaryReader reader);
+        protected abstract void SerializeWithoutType(BinaryWriter writer);
+
+        public void Deserialize(BinaryReader reader)
+        {
+            var usage = (TransactionAttributeUsage)reader.ReadByte();
+            if (Usage != usage) throw new FormatException();
+            DeserializeWithoutType(reader);
+        }
+
+        public void Serialize(BinaryWriter writer)
         {
             writer.Write((byte)Usage);
-            writer.WriteVarBytes(Data);
+            SerializeWithoutType(writer);
         }
 
-        public JObject ToJson()
+        public virtual JObject ToJson()
         {
             JObject json = new JObject();
             json["usage"] = Usage;
-            json["data"] = Convert.ToBase64String(Data);
             return json;
         }
 
         public static TransactionAttribute FromJson(JObject json)
         {
-            TransactionAttribute transactionAttribute = new TransactionAttribute();
-            transactionAttribute.Usage = (TransactionAttributeUsage)(byte.Parse(json["usage"].AsString()));
-            transactionAttribute.Data = Convert.FromBase64String(json["data"].AsString());
-            return transactionAttribute;
+            var usage = (TransactionAttributeUsage)(byte.Parse(json["usage"].AsString()));
+
+            return usage switch
+            {
+                TransactionAttributeUsage.OracleExpectedResult => OracleExpectedResult.FromJson(json),
+                _ => throw new FormatException(),
+            };
+        }
+
+        public static TransactionAttribute DeserializeFrom(BinaryReader reader)
+        {
+            var usage = (TransactionAttributeUsage)reader.ReadByte();
+
+            TransactionAttribute attrib;
+            switch (usage)
+            {
+                case TransactionAttributeUsage.OracleExpectedResult:
+                    {
+                        attrib = new OracleExpectedResult();
+                        break;
+                    }
+                default: throw new FormatException();
+            }
```

Comment:
Use `Activator.CreateInstance()`?

Reply:
I followed the same pattern as `NodeCapability`